lib/ExpensiMark.js

```diff
@@ -476,9 +470,29 @@ export default class ExpensiMark {
             }
             replacedText = replacedText.concat(textToCheck.substr(startIndex, (match.index - startIndex)));
 
+            // We want to avoid matching domains in email addresses so we don't render them as URLs.
+            // If the matched Url has a leading @ sign, The Url's domain can be a domain of email address.
+            // At this case, The matched string should not be parsed as Url,
+            // But it can contain another Url which should be parsed. (e.g. test@expensify.com/https://www.test.com)
+            // So We will retry to apply autolink rule to the string(e.g. /https://www.test.com) except for domain(e.g. expensify.com) after @ sign.
+            let abort = false;
+            let isReparsedByAutolink = false;
+
+            if ((match.index !== 0) && (textToCheck[match.index - 1] === '@')) {
+                const domainRegex = new RegExp('^(([a-z-0-9]+\\.)+[a-z]{2,})(\\S*)', 'i');
+                const domainMatch = domainRegex.exec(match[2]);
+
+                if ((domainMatch !== null) && (domainMatch[3] !== '')) {
+                    isReparsedByAutolink = true;
+                    replacedText = replacedText.concat(domainMatch[1] + this.replace(domainMatch[3], {filterRules: ['autolink']}));
+                } else {
+                    abort = true;
+                }
+            }
+
             if (abort || match[1].includes('<pre>')) {
                 replacedText = replacedText.concat(textToCheck.substr(match.index, (match[0].length)));
```

Comment:
I think this is still hard to understand, so here's a suggestion to make things clearer, but I'm not sure if what I'm saying in the comments, or if the updated variable names are accurate, so I could use your help
```js
            // We want to avoid matching domains in email addresses so we don't render them as URLs,
            // but we need to check if there are valid URLs after the email address and render them accordingly,
            // e.g. test@expensify.com/https://www.test.com
            let isDoneMatching = false;
            let shouldApplyAutoLinkAgain = true;

            // If we find a URL with a leading @ sign, we need look for other domains in the rest of the string
            if ((match.index !== 0) && (textToCheck[match.index - 1] === '@')) {
                const domainRegex = new RegExp('^(([a-z-0-9]+\\.)+[a-z]{2,})(\\S*)', 'i');
                const domainMatch = domainRegex.exec(match[2]);

                // If we find another domain, we apply the auto link rule again and set a flag to avoid doing it
                if ((domainMatch !== null) && (domainMatch[3] !== '')) {
                    replacedText = replacedText.concat(domainMatch[1] + this.replace(domainMatch[3], {filterRules: ['autolink']}));
                    shouldApplyAutoLinkAgain = false;
                } else {
                    // Otherwise, we're done applying rules
                    isDoneMatching = true;
                }
            }

            // Here we should probably explain in words what the code is doing, why, and what the condition means
            if (isDoneMatching || match[1].includes('<pre>')) {
                replacedText = replacedText.concat(textToCheck.substr(match.index, (match[0].length)));
            } else if (shouldApplyAutoLinkAgain) {
```

Reply:
> // Here we should probably explain in words what the code is doing, why, and what the condition means

it's out of scope for this PR.

> // If we find a URL with a leading @ sign, we need look for other domains in the rest of the string

If we find a URL with a leading @ sign, we need look for other valid URL in the rest of the string.

> // If we find another domain, we apply the auto link rule again and set a flag to avoid doing it

If we find another string trailing domain, we apply the auto link rule again and set a flag to avoid doing it

I amended something on your commenst, I think it's more accurate to explain

Reply:
> it's out of scope for this PR.

I understand you didn't write this in this PR, but we usually improve code around the code we're editing as we go, and given this isn't super clear, I think we should update it in this PR

> If we find a URL with a leading @ sign, we need look for other valid URL in the rest of the string.

Small correction: If we find a URL with a leading @ sign, we need to look for other valid URLs in the rest of the string.

> If we find another string trailing domain, we apply the auto link rule again and set a flag to avoid doing it

I don't think "string trailing domain" makes sense, how about

If we find another domain in the remainder of the string, we apply the auto link rule again and set a flag to avoid re-doing below

Reply:
> I understand you didn't write this in this PR, but we usually improve code around the code we're editing as we go, and given this isn't super clear, I think we should update it in this PR

for the condition `match[1].includes('<pre>')`, we can comment like below.

// We don't want to apply link rule if `match[1]` contains the code block inside the [] of the markdown e.g. `[```example```](https://example.com)`

other conditions's flags are already explained.
What the codes do can be figured out without comments, first one is to concatenate string without applying rule, second one is to apply rule

Reply:
> If we find another domain in the remainder of the string, we apply the auto link rule again and set a flag to avoid re-doing below

it's good

Reply:
Looks good to me 👍